engine/docker/docker.go

```diff
@@ -183,24 +184,44 @@ func (e *Docker) Destroy(ctx context.Context, pipelineConfig *spec.PipelineConfi
 
 // Run runs the pipeline step.
 func (e *Docker) Run(ctx context.Context, pipelineConfig *spec.PipelineConfig, step *spec.Step,
-	output io.Writer) (*runtime.State, error) {
+	output io.Writer, isDrone bool) (*runtime.State, error) {
```

Comment:
Do we need `isDrone` here? Ideally by this point the behaviour of both drone and harness should be same because the state of the pipeline, step etc is an intermediate representation.
Do we need to have a different behaviour for drone/harness here?

Reply:
Actually I don't want to change the drone behaviour unless needed. Since this only came up for Harness, I added the check.